src/storage/store/engine_tests.rs

```diff
@@ -541,6 +542,47 @@ mod tests {
         assert_eq!(0, verification_result.unwrap().messages_bytes.len());
     }
 
+    #[tokio::test]
+    async fn test_commit_username_proof_messages() {
+        let timestamp = messages_factory::farcaster_time();
+        let (mut engine, _tmpdir) = new_engine();
+        let name = "username".to_string();
+        let owner = "owner".to_string();
+        let signature = "signature".to_string();
+
+        register_user(FID_FOR_TEST, &mut engine).await;
+
+        let username_proof_add = messages_factory::username_proof::create_username_proof(
+            FID_FOR_TEST as u64,
+            crate::proto::username_proof::UserNameType::UsernameTypeFname,
+            name.clone(),
+            owner.clone(),
+            signature.clone(),
+            timestamp as u64,
+            None,
+        );
+
+        commit_message(&mut engine, &username_proof_add).await;
+
+        {
+            let username_proof_result = engine.get_username_proofs_by_fid(FID2_FOR_TEST);
+            assert!(username_proof_result.is_ok());
+
+            let messages_bytes_len = username_proof_result.unwrap().messages_bytes.len();
+            assert_eq!(0, messages_bytes_len);
+        }
+        {
+            let username_proof_result = engine.get_username_proofs_by_fid(FID_FOR_TEST);
+            assert!(username_proof_result.is_ok());
+
+            let messages_bytes_len = username_proof_result.unwrap().messages_bytes.len();
+            assert_eq!(1, messages_bytes_len);
+        }
+
+        // TODO: test get_username_proof (by name)
```

Comment:
do we need these two things? if so, now?

Reply:
No, we can skip for now. I think we might need it for RPC api compatibility, but we can add it then.